src/DipoleAlgorithms.cc

```diff
@@ -274,107 +212,42 @@ class NaiveDipoleFootprinter : public afw::detection::FootprintFunctor<MaskedIma
     source.set(_numNegativeKey, functor.getNumNegative());
-    source.set(getNegativeKeys().flag, false);
 }
 
-LSST_MEAS_ALGORITHM_PRIVATE_IMPLEMENTATION(NaiveDipoleFlux);
-
-PTR(meas::algorithms::AlgorithmControl) NaiveDipoleFluxControl::_clone() const {
-    return boost::make_shared<NaiveDipoleFluxControl>(*this);
-}
-
-PTR(meas::algorithms::Algorithm) NaiveDipoleFluxControl::_makeAlgorithm(
-    afw::table::Schema & schema,
-    PTR(daf::base::PropertyList) const &
-) const {
-    return boost::make_shared<NaiveDipoleFlux>(*this, boost::ref(schema));
+void NaiveDipoleFlux::fail(lsst::afw::table::SourceRecord & measRecord, lsst::meas::base::MeasurementError * error) const {
+    _flagHandler.handleFailure(measRecord, error);
 }
 
 
-
-/**
- * Implementation of Psf dipole flux
- */
-class PsfDipoleFlux : public DipoleFluxAlgorithm {
-public:
-
-    PsfDipoleFlux(PsfDipoleFluxControl const & ctrl, afw::table::Schema & schema) :
-        DipoleFluxAlgorithm(ctrl, schema, "jointly fitted psf flux counts"),
-        _chi2dofKey(schema.addField<float>(ctrl.name+".chi2dof", 
-                                           "chi2 per degree of freedom of fit")),
-        _avgCentroid(
-            addCentroidFields(schema, ctrl.name+".centroid", 
-                              "average of the postive and negative lobe positions")),
-        _negCentroid(
-            addCentroidFields(schema, ctrl.name+".neg.centroid", 
-                              "psf fitted center of negative lobe")),
-        _posCentroid(
-            addCentroidFields(schema, ctrl.name+".pos.centroid", 
-                              "psf fitted center of positive lobe")),
-        _flagMaxPixelsKey(schema.addField<afw::table::Flag>(ctrl.name+".flags.maxpix",
-                                                            "set if too large a footprint was sent to the algorithm"))
-    {}
-    template <typename PixelT>
-    std::pair<double,int> chi2(afw::table::SourceRecord & source,
-                afw::image::Exposure<PixelT> const & exposure,
-                double negCenterX, double negCenterY, double negFlux,
-                double posCenterX, double poCenterY, double posFlux
-                ) const;
-
-
-private:
-    template <typename PixelT>
-    void _apply(
-        afw::table::SourceRecord & source,
-        afw::image::Exposure<PixelT> const & exposure,
-        afw::geom::Point2D const & center
-    ) const;
-
-    LSST_MEAS_ALGORITHM_PRIVATE_INTERFACE(PsfDipoleFlux);
-
-    afw::table::Key<float> _chi2dofKey;
-    afw::table::KeyTuple<afw::table::Centroid> _avgCentroid;
-    afw::table::KeyTuple<afw::table::Centroid> _negCentroid;
-    afw::table::KeyTuple<afw::table::Centroid> _posCentroid;
-
-    afw::table::Key< afw::table::Flag > _flagMaxPixelsKey;
-};
-
 /**
  * Class to minimize PsfDipoleFlux; this is the object that Minuit minimizes
  */
-template<typename PixelT>
 class MinimizeDipoleChi2 : public ROOT::Minuit2::FCNBase {
```

Comment:
Just out of curiosity, why did you remove the template?  It seems fairly likely that float and double would both be useful, but maybe not.